.htaccess

```diff
@@ -1,240 +1,561 @@
+	AddOutputFilterByType DEFLATE text/html text/plain text/xml text/css text/javascript
 
-    # Netscape 4.06-4.08 have some more problems
-    #BrowserMatch ^Mozilla/4\.0[678] no-gzip
+############################################
+## Do not compress images
+## Apache Module mod_setenvif must be enabled
 
-    # MSIE masquerades as Netscape, but it is fine
-    #BrowserMatch \bMSIE !no-gzip !gzip-only-text/html
+	SetEnvIfNoCase Request_URI \.(?:gif|jpe?g|png)$ no-gzip dont-vary
 
-    # Don't compress images
-    #SetEnvIfNoCase Request_URI \.(?:gif|jpe?g|png)$ no-gzip dont-vary
+############################################
+## Please make sure proxies don't deliver the wrong content
+## Apache Module mod_headers must be enabled
 
-    # Make sure proxies don't deliver the wrong content
-    #Header append Vary User-Agent env=!dont-vary
+	#Header append Vary User-Agent env=!dont-vary
 
 </IfModule>
 
-<IfModule mod_ssl.c>
+<IfModule mod_expires.c>
 
 ############################################
-## make HTTPS env vars available for CGI mode
+## Control the setting of the Expires HTTP header
+## http://developer.yahoo.com/performance/rules.html#expires
 
-    SSLOptions StdEnvVars
+	ExpiresActive On
+
+	ExpiresByType audio/midi "access plus 1 month"
+	ExpiresByType audio/mpeg "access plus 1 month"
+	ExpiresByType audio/ogg "access plus 1 month"
+	ExpiresByType audio/wav "access plus 1 month"
+	ExpiresByType audio/webm "access plus 1 month"
+	ExpiresByType application/pdf "access plus 1 month"
+	ExpiresByType application/zip "access plus 1 month"
+	ExpiresByType font/woff "access plus 1 year"
+	ExpiresByType font/woff2 "access plus 1 year"
+	ExpiresByType image/bmp "access plus 1 year"
+	ExpiresByType image/gif "access plus 1 year"
+	ExpiresByType image/jpeg "access plus 1 year"
+	ExpiresByType image/png "access plus 1 year"
+	ExpiresByType image/svg+xml "access plus 1 year"
+	ExpiresByType image/tiff "access plus 1 year"
+	ExpiresByType image/webp "access plus 1 year"
+	ExpiresByType image/vnd.microsoft.icon "access plus 1 year"
+	ExpiresByType text/css "access plus 1 month"
+	ExpiresByType text/csv "access plus 1 month"
+	ExpiresByType text/html "access plus 1 month"
+	ExpiresByType text/javascript "access plus 1 month"
+	ExpiresByType text/plain "access plus 1 month"
+	ExpiresByType video/mp4 "access plus 1 month"
+	ExpiresByType video/mpeg "access plus 1 month"
+	ExpiresByType video/ogg "access plus 1 month"
+	ExpiresByType video/x-msvideo "access plus 1 month"
+
+	ExpiresDefault "access plus 2 days"
 
 </IfModule>
 
-<IfModule mod_rewrite.c>
+<IfModule mod_headers.c>
 
 ############################################
-## enable rewrites
+## Inform a client that the domain must only be accessed using HTTPS protocol
+## https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security
 
-    Options +FollowSymLinks
-    RewriteEngine on
+	Header set Strict-Transport-Security "max-age=31536000" env=HTTPS
```

Comment:
I don't think this should ever be set by default, especially not to 1 year. It should only be set when you're sure you have your SSL certs set up correctly.

Reply:
If you search for this header with Google you will see how much ink was consumed until this final version was reached. In an SSLabs test if the page has that header set you will get a better score. An administrator who does not set a SSL certificate correctly cannot be named an administrator.

Reply:
Personally I use a value of 2 years as well as includeSubdomains and preload. But it’s dangerous to set until you’re sure your certs are working properly and all subpaths (for example reverse proxied services) are on https.

I don’t disagree that it’s good to set, I would just comment by default.

Reply:
Let's say we comment on it and add an explanatory phrase. If you don't know about it, then you will be disqualified in SSL tests. If you know about it, you activate it and leave it like that. I agree that there may be situations when a certificate is not set correctly when the automatic task is done. If you have a more elaborate proposal we can discuss it.

Reply:
IMO, It's especially dangerous to enable for a server admin who does not know what it does. I don't have any other more elaborate proposal other than to just have it commented by default.